### PR TITLE
Add support for Deploy Tokens

### PR DESCRIPTION
**Pull request by PhilippeVienne**
Solve #539 

Inside the PR, the whole pom.xml is modified but the intended change is only to upgrade the integration test gitlab version.

### CODE REVIEW

File: src/main/java/org/gitlab4j/api/DeployTokensApi.java [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+     *
+     * @param page the page to get
+     * @param perPage the number of deploy tokens per page
+     * @return the list of DeployToken in the specified range
+     * @throws GitLabApiException if any exception occurs
+     */
+    public List<DeployToken> getDeployTokens(int page, int perPage) throws GitLabApiException {
+        Response response = get(Response.Status.OK, getPageQueryParams(page, perPage), "deploy_tokens");
+        return (response.readEntity(new GenericType<List<DeployToken>>() {}));
+    }
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/DeployTokensApi.java]: We are deprecating the methods that take ```(int page, int perPage)```, please remove this method.

File: src/main/java/org/gitlab4j/api/DeployTokensApi.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+     * @param page the page to get
+     * @param perPage the number of deploy tokens per page
+     * @return the list of DeployToken in the specified range
+     * @throws GitLabApiException if any exception occurs
+     */
+    public List<DeployToken> getProjectDeployTokens(Object projectIdOrPath, int page, int perPage) throws GitLabApiException {
+        Response response = get(Response.Status.OK, getPageQueryParams(page, perPage),
+                "projects", getProjectIdOrPath(projectIdOrPath), "deploy_tokens");
+        return (response.readEntity(new GenericType<List<DeployToken>>() {}));
+    }
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/DeployTokensApi.java]: We are deprecating the methods that take ```(int page, int perPage)```, please remove this method.

File: src/main/java/org/gitlab4j/api/models/DeployTokenScope.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @JsonValue
+    public String toValue() {
+        return (enumHelper.toString(this));
+    }
+
+    @Override
+    public String toString() {
+        return (enumHelper.toString(this));
+    }
+}
```
Comment (gmessner) [inline src/main/java/org/gitlab4j/api/models/DeployTokenScope.java]: Move ```DeployTokenScope``` to ```org.gitlab4j.api.Constants```

**Comment by PhilippeVienne** [comment]
I think I have issue with CRLF LF line ending and my ide. Can you just rewrite the file pom.xml and the Constants by your own. I was not able to fix that on my side.

**Comment by gmessner** [comment]
@PhilippeVienne I'll revert pom.xml and update the GitLab version prior to release.

Thanks for making the changes.

**Comment by gmessner** [comment]
@PhilippeVienne I've redone your commit with ```pom.xml``` and ```Constants.java``` fixed.  Release 4.14.24 is now available with your enhancements.

**Pull request merged**